pkg/reconciler/mtbroker/trigger/trigger.go

```diff
@@ -50,21 +61,66 @@ const (
 	subscriptionDeleted       = "SubscriptionDeleted"
 )
 
-func (r *Reconciler) reconcileTrigger(ctx context.Context, b *eventingv1.Broker, t *eventingv1.Trigger, brokerTrigger *corev1.ObjectReference) error {
+type Reconciler struct {
+	eventingClientSet clientset.Interface
+	dynamicClientSet  dynamic.Interface
+
+	// listers index properties about resources
+	subscriptionLister messaginglisters.SubscriptionLister
+	brokerLister       eventinglisters.BrokerLister
+	triggerLister      eventinglisters.TriggerLister
+	configmapLister    corev1listers.ConfigMapLister
+
+	// Dynamic tracker to track KResources. In particular, it tracks the dependency between Triggers and Sources.
+	kresourceTracker duck.ListableTracker
+
+	// Dynamic tracker to track AddressableTypes. In particular, it tracks Trigger subscribers.
+	uriResolver *resolver.URIResolver
+	impl        *controller.Impl
+}
+
+func (r *Reconciler) ReconcileKind(ctx context.Context, t *eventingv1.Trigger) pkgreconciler.Event {
+	logging.FromContext(ctx).Infow("Reconciling", zap.Any("Trigger", t))
 	t.Status.InitializeConditions()
 
 	if t.DeletionTimestamp != nil {
 		// Everything is cleaned up by the garbage collector.
 		return nil
 	}
+	// Start tracking the broker
+	r.trackBroker(ctx, t)
```

Comment:
oh! I see, you are going to track the broker directly. That works too.

Reply:
I think you have a race condition for late bound triggers without any brokers in the cluster yet.